### ISSUE TITLE
Inaccurate Current Usage for Matrix > Asset fields and Matrix > Supertable > Asset fields

### ISSUE THREAD

**Issue by howells** [born05/craft-assetusage#35]
I might be missing something here but the "Current Usage" for practically all assets on a site that uses matrix fields extensively shows "Unused". The site is a multi-site set up but whether it's the default site, or another, doesn't seem to have an affect (I'm using 2.3.1).

Two examples here but this applies throughout -

Via an asset field within Supertable, within a Matrix:

<img width="1257" alt="CleanShot 2022-08-16 at 09 15 23@2x" src="https://user-images.githubusercontent.com/145849/184831587-33b6cabe-de26-4992-a358-df3d7edd8cf0.png">

<img width="1045" alt="CleanShot 2022-08-16 at 09 16 01@2x" src="https://user-images.githubusercontent.com/145849/184831773-80f5ebcf-ee3b-4941-b47c-c5308ff20d4b.png">

Via an asset field, within a Matrix:

<img width="1265" alt="CleanShot 2022-08-16 at 09 17 19@2x" src="https://user-images.githubusercontent.com/145849/184832026-e9054cde-9588-4f9b-a73c-70f338624746.png">

<img width="1069" alt="CleanShot 2022-08-16 at 09 17 52@2x" src="https://user-images.githubusercontent.com/145849/184832119-daf9ba5a-f2d2-4e02-bd75-cd0775a75b71.png">

**Comment by darylknight** [comment]
Yeah, this plugin has been completely unusable for a long time, for a number of reasons. False positives, false negatives, incompatibility with drafts, revisions etc. I'm not sure it's still actually maintained. Which is a shame, as this seems to be the only plugin providing something that all our clients want.

**Comment by roelvanhintum** [comment]
@darylknight the plugin is maintained, but the problems with detecting usage are how much queries we should perform per asset. If we query a lot per asset, the performance of the asset overview would be insanely bad. I did try numerous times to get better numbers, but it's either missing things or resulting in minutes of load time. Feel free to suggest improvements, if you see any.
